docker/entrypoint.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+# Fail early upon error
+set -eu
+
+info () {
+    echo "I: $@"
+}
+
+# This file will exist if we've initialized postgres
+stamp=/var/lib/postgresql/9.6/main/initialized.stamp
+
+# Ensure the user starting the container has provided a password
+if [ -z "$POSTGRES_PASSWORD" ]
+then
+    echo "Please export \${POSTGRES_PASSWORD}"
```

Comment:
I wonder if there could be a better way which would avoid leaking any sensitive information through the environment.
I did check that skulpt has no env vars exposed, but I guess there could be other ways to see/get to it. so ideally should be via some config file

Reply:
You are suggesting starting the database without having those variables exported? The postgres image is a fairly common "bread and butter" image, and providing the envars (in some format) is what is commonly done. We can assume if the server is compromised with the environment variables and the application, you've already lost, for the most part.

The only thing I've seen recently for Docker that allows secrets (for builds) is the --secrets flag, I had some good discussion [here](https://www.reddit.com/r/programming/comments/bcumt6/secure_secrets_in_docker_builds/) (warning, that's reddit, lol) about how it was different from a build arg. That said, I think the point of focus was removing any trace of the variable in the Docker history, or bash history.

Do you have any ideas or suggestions for how this might be done @yarikoptic, or are you thinking out loud?

Reply:
for postgres container it is probably ok, since no child, client-visible process will operate within postgres DB process. But here it is the entry point for the web service, and password is also below within cmdline of `python web2py.py ` invocation.  So if somehow someone gets ability to list environment variables of the process, or list of running processes -- password would leak.  How critical and possible is that -- not sure ;-)   so indeed somewhat thinking out loud with extra paranoia in the voice

Reply:
Good points @yarikoptic ! I'm not a security expert but I can see this concern. So you are saying that if there is some (web entrypoint) in the application that isn't quoted correctly, and (given python) someone could hypothetically pipe an `import os; print(os.environ.values())` into it, we would be toast.

Reply:
Something like that or some other way to inject some "read only" code snippet into that running webapp process.